rj_gameplay/stp/rc.py

```diff
@@ -5,38 +5,91 @@
 from typing import List
 
 import numpy as np
+import warnings
 
 RobotId = int
 
 
 class Robot:
-    """State of the robot. Pose: [x, y, theta]. Twist: [dx, dy, dtheta]. Properties are
+    """State of a robot. Pose: [x, y, theta]. Twist: [dx, dy, dtheta]. Properties are
     to enforce that instances of this class should not be mutated."""
 
-    __slots__ = ["__id", "__pose", "__twist", "__has_ball"]
+    __slots__ = [
+        "__id", "__is_ours", "__pose", "__twist", "__ball_sense_triggered",
+        "__visible", "__has_ball_sense", "__kicker_charged",
+        "__kicker_healthy", "__lethal_fault"
+    ]
+
     __id: RobotId
+    __is_ours: bool
     __pose: np.ndarray
     __twist: np.ndarray
-    __has_ball: bool
-
-    def __init__(
-        self, robot_id: RobotId, pose: np.ndarray, twist: np.ndarray, has_ball: bool
-    ):
+    __visible: bool
+    __ball_sense_triggered: bool
+    __has_ball_sense: bool
+    __kicker_charged: bool
+    __kicker_healthy: bool
+    __lethal_fault: bool
+
+    def __init__(self, robot_id: RobotId, is_ours: bool, pose: np.ndarray,
+                 twist: np.ndarray, visible: bool, ball_sense_triggered: bool,
+                 has_ball_sense: bool, kicker_charged: bool,
+                 kicker_healthy: bool, lethal_fault: bool):
         """
         :param robot_id: Shell id of the robot.
+        :param is_ours: Whether the robot is one of our robots
         :param pose: Pose of the Robot. [x, y, theta].
         :param twist: Twist of the robot. [dx, dy, dtheta].
-        :param has_ball: Whether the robot has the ball or not.
+        :param ball_sense_triggered: Whether the ball sensor is triggered
+        :param visible: Whether the robot is being seen by the global vision system
+        :param has_ball_sense: Whether the robots ball sensor is functional
+        :param kicker_charged: Whether the robots kicker capacitors are charged
+        :param kicker_healthy: Whether the robots kicker is "healthy"
+        :param lethal_fault: Whether the robot has experienced a fault that will prevent it from functioning
         """
         self.__id = robot_id
+        self.__is_ours = is_ours
         self.__pose = pose
         self.__twist = twist
-        self.__has_ball = has_ball
+        self.__ball_sense_triggered = ball_sense_triggered
+        self.__visible = visible
+        self.__has_ball_sense = has_ball_sense
+        self.__kicker_charged = kicker_charged
+        self.__kicker_healthy = kicker_healthy
+        self.__lethal_fault = lethal_fault
+
+    @classmethod
+    def generate_test_robot(cls,
```

Comment:
This should honestly be in a separate module (i.e. `stp.testing.generate_test_robot()`), not a class function.

Reply:
Yeah, I can agree with that.